Content.Server/Nyanotrasen/Shipyard/Systems/ShipyardSystem.cs

```diff
@@ -0,0 +1,202 @@
+using Content.Server.Shuttles.Systems;
+using Content.Server.Shuttles.Components;
+using Content.Server.Station.Components;
+using Content.Server.Cargo.Systems;
+using Content.Server.Station.Systems;
+using Content.Shared.Shipyard;
+using Content.Server.Shipyard.Components;
+using Content.Shared.MobState.Components;
+using Content.Shared.GameTicking;
+using Robust.Server.GameObjects;
+using Robust.Server.Maps;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+
+namespace Content.Server.Shipyard.Systems
+{
+
+    public sealed partial class ShipyardSystem : SharedShipyardSystem
+    {
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly PricingSystem _pricing = default!;
+        [Dependency] private readonly ShuttleSystem _shuttle = default!;
+        [Dependency] private readonly StationSystem _station = default!;
+        [Dependency] private readonly CargoSystem _cargo = default!;
+        [Dependency] private readonly MapLoaderSystem _map = default!;
+        [Dependency] private readonly ShipyardConsoleSystem _shipyardConsole = default!;
+
+        public MapId? ShipyardMap { get; private set; }
+        private float _shuttleIndex;
+        private const float ShuttleSpawnBuffer = 1f;
+        private ISawmill _sawmill = default!;
+
+        public override void Initialize()
+        {
+            _sawmill = Logger.GetSawmill("shipyard");
+            _shipyardConsole.InitializeConsole();
+            SubscribeLocalEvent<ShipyardConsoleComponent, ComponentInit>(OnShipyardStartup);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRoundRestart);
+        }
+
+        private void OnShipyardStartup(EntityUid uid, ShipyardConsoleComponent component, ComponentInit args)
+        {
+            SetupShipyard();
+        }
+
+        private void OnRoundRestart(RoundRestartCleanupEvent ev)
+        {
+            CleanupShipyard();
+        }
+
+        /// <summary>
+        /// Adds a ship to the shipyard, calculates its price, and attempts to ftl-dock it to the given station
+        /// </summary>
+        /// <param name="stationUid">The ID of the station to dock the shuttle to</param>
+        /// <param name="shuttlePath">The path to the grid file to load. Must be a grid file!</param>
+        public void PurchaseShuttle(EntityUid? stationUid, string shuttlePath, out ShuttleComponent? vessel)
+        {
+            if (!TryComp<StationDataComponent>(stationUid, out var stationData) || !TryComp<ShuttleComponent>(AddShuttle(shuttlePath), out var shuttle))
+            {
+                vessel = null;
+                return;
+            }
```

Comment:
Yes, these would be loaded into the shipyard map.

The map file needs to be loaded and init before we can get any proper values from it.  If the map file successfully loads but it's not a shuttle, it will refuse to try to bring it onto the playable map.  This is to prevent even an admin from accidentally loading say, a station or salvage wreck file and the system trying to dock stations together, etc.

I could add more into the AddShuttle method to remove any erroneous grids that have been loaded, but since I am forced to use TryLoad (sloth almost murdered me for trying to use loadgrid) there are any number of things that could be loaded into the shipyard.

Essentially this all came about because I can't use loadgrid anymore

The shipyard map gets cleaned up at the end of the round.